guides/doc-Provisioning_Guide/topics/Resources.adoc

```diff
@@ -259,10 +259,34 @@ PXE-based templates that deploy to the template {SmartProxy} associated with a s
 
 *Finish*
 
-Post-configuration scripts to use when the main provisioning process completes. This is completed as an SSH task. You can use Finishing templates only for imaged-based provisioning in virtual environments. Do not confuse an image with a foreman discovery ISO, which is sometimes called a Foreman discovery image. An image in this context is an install image in a virtualized environment for easy deployment.
+Post-configuration scripts to execute using an SSH connection when the main provisioning process completes. You can use Finishing templates only for imaged-based provisioning in virtual or cloud environments that do not support user_data. Do not confuse an image with a foreman discovery ISO, which is sometimes called a Foreman discovery image. An image in this context is an install image in a virtualized environment for easy deployment.
 
 When a finish script successfully exits with the return code `0`, {ProjectName} treats the code as a success and the host exits the build mode. Note that there are a few finish scripts with a build mode that uses a _call back_ HTTP call. These scripts are not used for image-based provisioning, but for post configuration of operating-system installations such as Debian, Ubuntu, and BSD.
 
+*user_data*
+
+Post-configuration scripts for providers that accept custom data, also known as seed data. This template does not require {Project} to be able to reach the host, the cloud or virtualization platform is responsible for delivering the data to the image.
+
+Ensure that you meet the following requirements to use the `user_data` template:
+
+* You can use the user_data template to provision virtual machines in cloud or virtualised environments only.
+* Ensure that the image that you want to provision supports the user_data template.
+* On a host, install the software to read the data and set it to start during boot. For example, `cloud-init`, which expects YAML input, or `ignition`, which expects JSON output. To install the software, you can download it over HTTP, or mount a floppy image, or use any other technique.
+
+Some platforms, for example VMWare, utilise its own way and format for seeding images; you cannot use cloud-init YAML input in these cases. There is a special template kind named `cloud-init` to directly integrate with cloud-init client. For more information, see xref:Provisioning_Virtual_Machines_in_VMware_vSphere-Provisioning_with_cloudinit_and_userdata_templates[].
```

Comment:
Do we need this paragraph? It seems pretty similar to the cloud-init paragraph below.